src/sentry/tagstore/snuba/backend.py

```diff
@@ -586,13 +587,21 @@ def get_tag_value_paginator_for_projects(
 
         conditions = []
 
-        if snuba_key in BLACKLISTED_COLUMNS:
-            snuba_key = "tags[%s]" % (key,)
-
-        if query:
-            conditions.append([snuba_key, "LIKE", u"%{}%".format(query)])
+        if key in SearchVisitor.numeric_keys and snuba_key not in BLACKLISTED_COLUMNS:
+            converted_query = int(query) if query is not None and query.isdigit() else None
+            if converted_query is not None:
+                conditions.append([snuba_key, ">=", converted_query - 50])
+                conditions.append([snuba_key, "<=", converted_query + 50])
+            else:
+                conditions.append([[snuba_key, ">=", 0], [snuba_key, "<", 0]])
```

Comment:
If the converted_query is None, do we need to add a condition on this field at all? Maybe we can check for not null? If we're just trying to get all results, I think no condition would do the same thing.

Reply:
:+1: Good call, looks like no condition does the same thing, will update. 🤔 seems to be the same for the string values too though, I wonder why we did `key != ""`